src/components/common/GProfiler/reducer.ts

```diff
@@ -0,0 +1,64 @@
+import produce, { Draft } from 'immer';
+
+import * as t from './actionTypes';
+import { State } from './model';
+import { GProfilerAction } from './actions';
+import { Reducer } from 'react';
+
+const initialState: State = {
+    error: '',
+    display: false,
+    selectedTopGeneListsSizes: [],
+    isFetchingAvailableOrganisms: false,
+    availableOrganisms: [],
+    selectedOrganism: '',
+    selectedSortBy: '',
+    gProfilerToken: '',
+};
+
+const reducer: Reducer<State, GProfilerAction> = produce(
```

Comment:
This is an unfortunate thing about TypeScript, it's structurally typed so this is "correct". But `Reducer` is nominally incorrect (and confusing) here. [Here](https://github.com/reduxjs/redux/blob/master/src/types/reducers.ts#L29) is the redux `Reducer` type definition. If you want to properly type annotate this, be my guest (but with the redux `Reducer` type) or just don't type annotate the `reduce()` function.

Reply:
Sorry I'm not following why this is incorrect
It states: 
```
 * @template S The type of state consumed and produced by this reducer.
 * @template A The type of actions the reducer can potentially respond to.
```
This what I provide and it should return a Reducer type no ?

Reply:
Yeah this is the confusing part! React **and** Redux define a `Reducer` type. The reducer here is the Redux one, but you're importing/using the React one.

After I saw you typing this I went and found the correct type (https://github.com/aertslab/SCope/pull/381/files#diff-efbeaa2578aacd09fb3c964e4fbb2b4bR3) and used it in the cleanup PR.

Reply:
Oh I see, thanks
I will import the `redux` one